### ISSUE TITLE
4.1 and Liteqube

### ISSUE THREAD

**Issue by kennethrrosen** [mirage/qubes-mirage-firewall#174]
I manually unpacked and installed the mirage fw and kernel into a liteqube setup on 4.1 (I realize liteqube also offers it as its own fw). The VM seems to function, but it cannot supply firewall rules. There is a red banner notice reading "Net qube does not support 'qubes-firewall'. Firewall restrictions will not be applied."

**Comment by palainp** [comment]
I see this too, but it will work as expected (to validate, you can try to restrict to one domain and you won't be able to get traffic outside that domain).

The message may come from a missing signaling part in the communication protocol between Qubes and the unikernel. We can probably leave this issue open until this message is fixed.

**Comment by kennethrrosen** [comment]
I did this at install, as per the README file instructions. No joy.

**Comment by palainp** [comment]
Would you mind to give the result of `qvm-firewall AppVM list`?

**Comment by kennethrrosen** [comment]
[user@dom0 ~]$ qvm-firewall fw-mirage list
NO  ACTION  HOST  PROTOCOL  PORT(S)  SPECIAL TARGET  ICMP TYPE  EXPIRE  COMMENT
0   accept  -     udp       1194     -               -          -       -
1   accept  -     tcp       5995     -               -          -       -
2   accept  -     tcp       8443     -               -          -       -
3   accept  -     udp       4569     -               -          -       -
4   accept  -     udp       5060     -               -          -       -
5   accept  -     -         -        dns             -          -       -
6   accept  -     icmp      -        -               -          -       -
7   drop    -     -         -        -               -          -       -

**Comment by palainp** [comment]
Thank you for that output.

These rules (set on mirage-fw in Qubes Manager but not dealt inside mirage-fw, it's up to it's netvm to filter the netflow) should be visible in the netvm with `iptables -L` (it may worth to check there if something is missing), and I agree that any flow not matching an accept line should be dropped in netvm.

It may also worth to try to set the rules on your vpn VM tab in Qubes Manager to check how mirage-fw deals with that.

**Comment by kennethrrosen** [comment]
They don't seem to populate in core-net, but of course the rules work when set through the Qubes Manager in sys-vpn, though I'm stumped as to why I can't set all necessary rules in fw-mirage.

**Comment by palainp** [comment]
Yes I think you should be able to restrict the vpn netflow by setting rules on the mirage-fw tab in Qubes Manager. However this leads into rules being checked by sys-net or core-net which is untrusted, and that sounds odd to me.

Given it seems to work as intended with a vanilla Qubes, is it possible to try with liteqube and a normal linux firewall: setting rules on that VM and see if it works as vpn killswitch? If it works we should fix something into mirage-fw, and if not it's probably something relevant in liteqube (and we may open an issue there with this one as background).

**Comment by kennethrrosen** [comment]
I created a new default sys-firewall `qubesctl state.apply qvm.sys-firewall` and attached the qubes: core-net <--> sys-firewall <---> sys-vpn <---> appqube and applied the firewall rules to sys-firewall. This worked as intended. 

When I set the rules in sys-vpn, the appqube, and mirage-fw, those rules don't appear to be visible in core-net.

**Comment by palainp** [comment]
To summarize the tests, am I right with the following?
- standard Qubes, firewall rules on mirage-fw: can limit traffic to vpn only
- standard Qubes, firewall rules on sys-firewall: can limit traffic to vpn only
- Qubeslite, firewall rules on mirage-fw: *let vpn and non vpn traffic to go through*
- Qubeslite, firewall rules on sys-firewall: can limit traffic to vpn only
- Qubeslite, firewall rules on sys-vpn with mirage-fw as netvm: can limit traffic to vpn only

So the situation is that we can't filter traffic outside vpn with mirage-fw directly after core-net, but we can't reproduce it with standard Qubes distribution.

**Comment by palainp** [comment]
Hi, I found an easy way to check firewall rules in core-net, would you mind to type:
- before applying fw rules to qubes-mirage-fw
- after applying fw rules to qubes-mirage-fw
- before applying fw rules to sys-firewall
- after applying fw rules to sys-firewall

You should get something referring `chain qbs-10-137-0-XX` which stands for the IP the qubes and see the rules there.

**Comment by palainp** [comment]
Oh that's probably because liteqube uses the `-minimal` templates. That may (or may not) be a root cause for this issue.

I'm also thinking about the vpn script (https://github.com/a-barinov/liteqube/blob/main/4.VPN/default/debian-core/etc/protect/template.core-vpn/usrlocal/bin/liteqube-vpn), please, if that's what you use as vpn, can you check that the iptables rules are correctly up ? (this should be visible in the core-vpn VM with `sudo iptables -L -v`).

**Comment by kennethrrosen** [comment]
I don't use the liteqube VPN, but rather a variant of this: https://forum.qubes-os.org/t/how-to-setup-openvpn-fedora-appvm-for-ovpn/3354

**Comment by palainp** [comment]
So I have exactly the same (classical Qubes, fedora sys-net). So there's probably a way to check into core-net the filter rules you set for mirage-fw. Sadly I didn't find something useful yet :(

**Comment by palainp** [comment]
@arkenoi, @froeb I recently saw that you've been trying to adapt liteqube to Qubes 4.2.

I'm throwing a bottle into the sea in case you might have an idea of what could cause issues with mirage-fw (not) acting as a killswitch when used in a VPN context.
TLDR: on a classic Qubes 4.2 mirage-fw correctly acts as a killswitch as well as sys-fw, on liteqube mirage-fw doesn't but core-fw does.

**Comment by palainp** [comment]
@arkenoi you seem to have fully updated to Qubes 4.2 recently, do you know if this issue is still present in latest Qubes release?

**Comment by arkenoi** [comment]
I did not try yet, but I will.

**Comment by palainp** [comment]
Dear @arkenoi , apart the recent issue with windows guests, do you know if the issue is solved with current liteqube?

**Comment by arkenoi** [comment]
apparently it is, but did not test deeply. could you suggest a proper test protocol to consider the issue closed? can ask clanker, of course..

**Comment by palainp** [comment]
The issue is about having killswitch rules. On a classic Qubes 4.2 mirage-fw correctly acts as a killswitch as well as sys-fw, on liteqube mirage-fw doesn't but core-fw does.

**Comment by arkenoi** [comment]
Could you point me to exact ruleset to test?

**Comment by palainp** [comment]
I would try to use any VPN provider, restrict the VPN AppVM flows to the address of the VPN server with qvm-firewall. The reproduction case is that normla traffic (e.g. when the VPN client is not connected or traffic is not properly routed) passes with mirage-fw and does not wth linux-fw.
